app/code/Magento/Sitemap/Controller/Adminhtml/Sitemap/Save.php

```diff
@@ -6,11 +6,70 @@
 namespace Magento\Sitemap\Controller\Adminhtml\Sitemap;
 
 use Magento\Backend\App\Action;
+use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\App\Filesystem\DirectoryList;
 use Magento\Framework\Controller;
 
-class Save extends \Magento\Sitemap\Controller\Adminhtml\Sitemap
+/**
+ * Save XML sitemap action.
+ */
+class Save extends \Magento\Sitemap\Controller\Adminhtml\Sitemap implements HttpPostActionInterface
 {
+    /**
+     * Maximum length of sitemap filename
+     */
+    const MAX_FILENAME_LENGTH = 32;
+
+    /**
+     * @var $_stringValidator
+     */
+    public $_stringValidator;
+
+    /**
+     * @var $_pathValidator
+     */
+    public $_pathValidator;
+
+    /**
+     * @var $_sitemapHelper
+     */
+    public $_sitemapHelper;
+
+    /**
+     * @var $_filesystem
+     */
+    public $_filesystem;
+
+    /**
+     * @var $_sitemapFactory
+     */
+    public $_sitemapFactory;
+
+    /**
+     * Save constructor.
+     * @param Action\Context $context
+     * @param \Magento\Framework\Validator\StringLength $stringValidator
+     * @param \Magento\MediaStorage\Model\File\Validator\AvailablePath $pathValidator
+     * @param \Magento\Sitemap\Helper\Data $sitemapHelper
+     * @param \Magento\Framework\Filesystem $filesystem
+     * @param \Magento\Sitemap\Model\SitemapFactory $sitemapFactory
+     */
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
```

Comment:
Please add dependencies in a backward compatible way, see "Adding a constructor parameter" section of [Backward compatible development guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php) for details

Reply:
okay I will update it.